service/gcs/bridge/bridge_unit_test.go

```diff
@@ -0,0 +1,619 @@
+package bridge
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"io"
+	"strings"
+	"sync"
+	"testing"
```

Comment:
I probably need to look at your tests more closely, but I feel like it should be fine to set handlers/errors/etc. which are more specific to individual tests deeper in the closure stack where those tests are. Like, put state that's only used by certain tests only in the scope for those tests. Ginkgo is very much intended for unit testing :). It also gets rid of a lot of the code duplication going on in the "testing" tests. I think in general we should try to use the same testing framework for the whole GCS if we can.